README.md

```diff
@@ -35,3 +35,97 @@ steps should work:
 
 You should now have a working `build/Release/CheriVis.app`.  For more complex
 build configurations or to edit the code, open the XCode project.
+
+### Building on Linux (Ubuntu 16.04.4 LTS Xenial Xerus)
+#### Cheritrace
+On Linux, first you must build the cheritrace library, e.g. by doing
+
+    $ cd cheritrace
+    $ mkdir Build
+    $ cd Build
+    $ # Make sure you have a recent clang
+    $ cmake .. -GNinja -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DCMAKE_C_COMPILER=/usr/bin/clang -DLLVM_CONFIG=/path/to/cheri/sdk/bin/llvm-config
+    $ ninja
+
+If you get a message about
+`/path/to/cheri/sdk/lib/libLLVMDebugInfoDWARF.a` being the wrong
+format (and if you do `file
+/path/to/cheri/sdk/lib/libLLVMDebugInfoDWARF.a` and it says it is
+empty, you need a CHERI-SDK which has that file, e.g. by using
+`cheribuild.py` (from github).
+
+
+#### Modern GNUstep (libobjc2)
+The Ubuntu GNUstep stuff is too old, so these steps install a GNUstep
+system under `/usr/GNUstep/`.
+
+    mkdir GNUstep-build
+    cd GNUstep-build
+
+    export CC=clang
+    export CXX=clang++
+
+    mkdir -p libobjc2 && wget -qO- https://github.com/gnustep/libobjc2/archive/v1.8.1.tar.gz | tar xz -C libobjc2 --strip-components=1
```

Comment:
You're probably better off building from git for libobjc2.  I really need to do a release...

Reply:
Okay, using the recent git versions of all of them 
```
    git clone 'https://github.com/gnustep/libobjc2.git'
    git clone 'https://github.com/gnustep/tools-make.git'
    git clone 'https://github.com/gnustep/libs-base.git'
    git clone 'https://github.com/gnustep/libs-gui.git'
    git clone 'https://github.com/gnustep/libs-back.git'
```
(just using `https://github.com/gnustep/libobjc2.git` seems to break things), but libs-base complains about
```
../Headers/GNUstepBase/GSVersionMacros.h:290:10: error: "You are building gnustep-base using the objc-nonfragile-abi but your gnustep-base was not configured to use it."
#        error "You are building gnustep-base using the objc-nonfragile-abi but your gnustep-base was not configured to use it."
```
even when I configure libs-base with `--disable-mixedabi` (though that only results in one error about nonfragile ABI, rather than about five).